lua/compe_tags/init.lua

```diff
@@ -0,0 +1,58 @@
+local compe = require("compe")
+local Source = {}
+
+function Source.new()
+  return setmetatable({}, { __index = Source })
+end
+
+function Source.get_metadata(_)
+  return {
+    priority = 90;
+    menu = '[Tag]';
+  }
+end
+
+function Source.determine(_, context)
+  return compe.helper.determine(context)
+end
+
+function Source.complete(_, context)
+  context.callback({
+    items = vim.fn.getcompletion(context.input, "tag"),
+    incomplete = true
+  })
+end
+
+function Source.documentation(_, context)
+  local tags = {}
+  local word = context.completed_item.word or ''
+
+  if word == '' then return context.abort() end
+
+  local slice = function(tbl, first, last, step)
+    local sliced = {}
+
+    for i = first or 1, last or #tbl, step or 1 do
+      sliced[#sliced+1] = tbl[i]
+    end
+
+    return sliced
+  end
+
+  tags = vim.tbl_map(function(item)
+    if not vim.tbl_contains(tags, item) then
+      return item.filename
+    end
+  end, vim.fn.taglist(word))
+
+  if #tags > 10 then
+    tags = table.insert(
+      slice(tags, 1, 9), string.format("...and %d more", #slice(tags, 10))
```

Comment:
I couldn't test this one so I don't know if this work or not.

Reply:
The below code more simple but this implementation is enough to me.

```lua
  local items = {}
  local tags = vim.fn.taglist(args.completed_item.word)
  for i, tag in ipairs(tags) do
    if 10 < i then
      table.insert(items, ('...and %d more'):format(#tags - 10))
      break
    end
    table.insert(items, tag.filename)
  end
```

Reply:
that looks simpler and better, thanks!